lab_3_generate_by_ngrams/main.py

```diff
@@ -331,6 +496,27 @@ def continue_sequence(
 
         In case of corrupt input arguments or unexpected behaviour of methods used return None.
         """
+        if (not isinstance(sequence, tuple) or not isinstance(next_tokens, list) or
+                not isinstance(sequence_candidates, dict) or not sequence):
+            return None
+        if (not next_tokens or not sequence_candidates or
+                sequence not in sequence_candidates or
+                len(next_tokens) > self._beam_width):
+            return None
+
+        new_sequence_candidates = {}
+        for key, value in sequence_candidates.items():
+            if key != sequence:
+                probability = value
+                for token, token_probability in next_tokens:
+                    new_sequence = key + (token,)
+                    new_probability = probability + (-1) * math.log(token_probability)
```

Comment:
why do you need the multiplication? can you just substitute log?

Reply:
changed it